sqlglot/generator.py

```diff
@@ -1411,7 +1413,7 @@ def lock_sql(self, expression: exp.Lock) -> str:
         return f"{lock_type}{expressions}{wait or ''}"
 
     def literal_sql(self, expression: exp.Literal) -> str:
-        text = expression.this or ""
+        text = expression.this if expression.this is not None else ""
```

Comment:
This one was converting literal `0` to `""`

Reply:
it shouldn't because this should always be a string, "0", and not the literal 0